### PR TITLE
Fix deadlock in SampleBuffer

### PR DESCRIPTION
**Pull request by Wallacoloo**
This fixes a possible deadlock scenario in `SampleBuffer::play()` caused by locking a lock and failing to unlock it before returning. 

In fixing this, I've opted to apply the much safer [RAII-style](http://www.cs.wustl.edu/~schmidt/PDF/ScopedLocking.pdf) of locking through the use of [QReadLocker](http://doc.qt.io/qt-4.8/qreadlocker.html) and [QWriteLocker](http://doc.qt.io/qt-4.8/qwritelocker.html) in places that previously manipulated the lock manually (No need to read the `RAII-style` link if you're already familiar with the concept of scoped locks). This will help to prevent errors like this from arising in the future.

While I was at it, I also applied some trivial formatting fixes (i.e. tabbing fixes).

### CODE REVIEW

File: src/core/SampleBuffer.cpp [hunk 1]
```diff
@@ -632,35 +621,25 @@ bool SampleBuffer::play( sampleFrame * _ab, handleState * _state,
...
 	{
-		if( play_frame >= endFrame )
+		if( play_frame >= endFrame || ( endFrame - play_frame ) / freq_factor == 0 )
 		{
-			m_varLock.unlock();
+			// the sample is done being played
 			return false;
 		}
-
-		if( ( endFrame - play_frame ) / freq_factor == 0 ) return false;
```
Comment (Wallacoloo) [inline src/core/SampleBuffer.cpp]: Deadlock was here.

**Comment by curlymorphic** [comment]
nice catch, 

Having read through the lock changes I feel this is an improvement and should be committed.

I am aware that in an ideal world we should be lockless, but we currently don't have the resources for that achievement.

**Comment by midi-pascal** [comment]
:+1:

**Comment by Wallacoloo** [comment]
Thanks for the feedback. Gonna go ahead and merge this now.

**Pull request merged**